src/tiled/newtilesetdialog.cpp

```diff
@@ -90,15 +88,15 @@ NewTilesetDialog::NewTilesetDialog(QWidget *parent) :
     connect(mUi->tilesetType, SIGNAL(currentIndexChanged(int)),
             SLOT(tilesetTypeChanged(int)));
     connect(mUi->dropperButton, SIGNAL(clicked(bool)), SLOT(pickColorFromImage()));
-    connect(mPopup, SIGNAL(colorSelected(QColor)), SLOT(colorSelected(QColor)));
     mUi->imageGroupBox->setVisible(tilesetType == 0);
     updateOkButton();
 }
 
 NewTilesetDialog::~NewTilesetDialog()
 {
     delete mUi;
-    delete mPopup;
+    if (mPopup != nullptr)
+        delete mPopup;
```

Comment:
Just a note, you never need to check a pointer before deleting it. The `delete` operator is by design safe to use on a null pointer.